playbooks/06-deploy-architecture.yml

```diff
@@ -66,42 +66,64 @@
-          ansible.builtin.set_fact:
-            cifmw_ci_gen_kustomize_values_ssh_authorizedkeys: >-
-              {{ lookup('file', '~/.ssh/id_cifw.pub', rstrip=False) }}
-            cifmw_ci_gen_kustomize_values_ssh_private_key: >-
-              {{ lookup('file', '~/.ssh/id_cifw', rstrip=False) }}
-            cifmw_ci_gen_kustomize_values_ssh_public_key: >-
-              {{ lookup('file', '~/.ssh/id_cifw.pub', rstrip=False) }}
-            cifmw_ci_gen_kustomize_values_migration_pub_key: >-
-              {{ lookup('file', _ssh_file ~ '.pub', rstrip=False)}}
-            cifmw_ci_gen_kustomize_values_migration_priv_key: >-
-              {{ lookup('file', _ssh_file, rstrip=False) }}
-            cifmw_ci_gen_kustomize_values_sshd_ranges: >-
+            _ifaces_vars: >-
               {{
-                [cifmw_networking_env_definition.networks.ctlplane.network_v4] +
-                (
-                  [
-                    _ipv4_network_data.network + '/' + _ipv4_network_data.prefix
-                  ]
-                ) if (_ipv4_network_data | length > 0) else []
+                _ctl_data.ansible_interfaces |
+                map('regex_replace', '^(.*)$', 'ansible_\1')
               }}
+            _controller_host: "{{ _ctl_data.ansible_host }}"
+          block:
+            - name: Generate needed facts out of local files
+              vars:
+                _ipv4_network_data: >-
+                  {{
+                    _ctl_data | dict2items |
+                    selectattr('key', 'in', _ifaces_vars) |
+                    selectattr('value.ipv4.address', 'defined') |
+                    selectattr('value.ipv4.address', 'equalto', _controller_host) |
+                    map(attribute='value.ipv4') | first | default({})
+                  }}
+              ansible.builtin.set_fact:
+                cifmw_ci_gen_kustomize_values_ssh_authorizedkeys: >-
+                  {{ lookup('file', '~/.ssh/id_cifw.pub', rstrip=False) }}
+                cifmw_ci_gen_kustomize_values_ssh_private_key: >-
+                  {{ lookup('file', '~/.ssh/id_cifw', rstrip=False) }}
+                cifmw_ci_gen_kustomize_values_ssh_public_key: >-
+                  {{ lookup('file', '~/.ssh/id_cifw.pub', rstrip=False) }}
+                cifmw_ci_gen_kustomize_values_migration_pub_key: >-
+                  {{ lookup('file', _ssh_file ~ '.pub', rstrip=False)}}
+                cifmw_ci_gen_kustomize_values_migration_priv_key: >-
+                  {{ lookup('file', _ssh_file, rstrip=False) }}
+                cifmw_ci_gen_kustomize_values_sshd_ranges: >-
+                  {{
+                    [cifmw_networking_env_definition.networks.ctlplane.network_v4] +
+                    (
+                      [
+                        _ipv4_network_data.network + '/' + _ipv4_network_data.prefix
+                      ]
+                    ) if (_ipv4_network_data | length > 0) else []
+                  }}
+          rescue:
```

Comment:
Nice to see `rescue`

https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_blocks.html#handling-errors-with-blocks